### PR TITLE
Skytap prototype

### PR DESCRIPTION
**Pull request by itsjeyd**
cf. [OC-2358](https://tasks.opencraft.com/browse/OC-2358)

### CODE REVIEW

File: xblock_skytap/skytap.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+import skytap as skytap_library
+
+from xblock.core import XBlock
+from xblock.fields import Scope, String
+from xblock.fragment import Fragment
+from xblockutils.resources import ResourceLoader
+from xblockutils.studio_editable import StudioEditableXBlockMixin
+
+from .utils import _
```
Comment (bdero) [inline xblock_skytap/skytap.py]: Nit: Would it be possible to do an absolute import?

File: xblock_skytap/templates/skytap.html [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        <option value="fr-be">French-Belgium</option>
+        <option value="fr-ch">French-Switzerland</option>
+        <option value="de">German</option>
+        <option value="de-ch">German-Switzerland</option>
+        <option value="is">Icelandic</option>
+        <option value="it">Italian</option>
+        <option value="jp">Japanese</option>
+        <option value="no">Norwegian</option>
+        <option value="pt">Polish</option>
+        <option value="es">Spanish</option>
```
Comment (bdero) [inline xblock_skytap/templates/skytap.html:21]: Would it be possible to fetch this info from the SkyTap API?

Comment (itsjeyd) [reply xblock_skytap/templates/skytap.html:21]: @bdero Nope, unfortunately not. From the API documentation (v1, v2) about the `vnc_keymap` setting:

> Keyboard language mapping. UTF-8 character type. For a list of supported international keyboards and attribute values, see [Setting an international keyboard layout for a VM](http://help.skytap.com/Setting_an_International_Keyboard_Layout.html). Attribute values are listed in parentheses on that page.

**Comment by bdero** [comment]
@itsjeyd I tried installing this into and running the workbench like this:
   ```
   pip install git+https://github.com/open-craft/xblock-skytap.git@skytap-prototype#egg=xblock-skytap==prototype1
   SKYTAP_TOKEN=nothing SKYTAP_USER=doesntexist ./manage.py runserver
   ```
But since there aren't any scenarios and it didn't look totally trivial to set that up, I installed it into a local devstack and populated `~/.skytap` with the settings:
```
pip install -e git+https://github.com/open-craft/xblock-skytap.git@skytap-prototype#egg=xblock-skytap==prototype1
```

However, when I fired up studio and tried to add 'skytap' to the advanced modules list for a course, it immediately reverts when I save without any useful output. Based on the entry point in `setup.py`, I think `skytap` is the name I'm supposed to use, but I also tried others for good measure, like `xblock_skytap`, `skytap_xblock`, `xblock-skytap`, etc.

Do you have any ideas about what I might be doing wrong?

**Comment by itsjeyd** [comment]
@bdero I don't think I ever ran into that particular problem. When I tested this last, I didn't put settings into `~/.skytap`, I just exported `SKYTAP_USER` and `SKYTAP_TOKEN` directly. But in any case, it doesn't make a whole lot of sense to test this now, since we'll be removing the `skytap` dependency as part of the first implementation story. We can't spend any additional time on the discovery story, since we're pretty much at the budget cap for that.

**Comment by bdero** [comment]
@itsjeyd My issue is actually not with the skytap library - the issue is that I can't actually enable to xblock at all, which is most likely because I'm not doing something right as I don't have a lot of experience with xblock dev. ;)

**Comment by bdero** [comment]
:+1:

- [x] I tested this: I installed the xblock into edxapp (`pip install -e git+https://github.com/open-craft/xblock-skytap.git@skytap-prototype#egg=xblock-skytap==prototype1`), enabled it in advanced settings on a new course, created a Skytap XBlock problem in a blank unit, and verified that the keyboard layout selection and launch button appear.
- [x] I read through the code
- [x] I checked for accessibility issues
- [ ] ~Includes documentation~ Not necessary yet - this is mostly just scaffolding of the XBlock.

**Comment by itsjeyd** [comment]
Thanks @bdero! Merging now.

**Pull request merged**